lib/docassemble/server_install.js

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env node
+
+const session_vars = require('../utils/session_vars');
+
+const server_install = async () => {
+  session_vars.save_install_type("server");
+};  // Ends server_install();
```

Comment:
This script is even weirder now that it doesn't do anything besides setting the install type for the rest of Kiln to use. Like, you can't look at the script and understand what's happening at all.

I'm not sure what's more hacky; leaving this script and the `session_vars.save_install_type` APIs, or just echoing a `{ "da_install_type": "server" }` to `runtime_config.json` directly in the GitHub action. But I'm not happy with the state of these APIs now. Now that `dainstall` is doing all of the work, maybe it is better to pass `install_type` through on env var, idk.

Reply:
We can't just `echo` because there's other stuff that needs to be in the `.json`. I know what you mean, though. I don't see a great solution either, though I do admit I like `dainstall`. Just one line, so much simpler.

Since `install_type` is going to be temporary and eventually taken over by passing a base url in for the command line, how much does it matter? Partly depends on how long we keep this stuff around I suppose.

It occurs to me that if we're later going to require the base url, that'll be another breaking change. Maybe we need to take care of that before we go implementing an intermediate measure like `install_type`. I think that may make my head explode. If it's the better way, though...

Reply:
Would love your thoughts on this, @BryceStevenWilley (sorry it's buried in all these comments). I've started looking at the command line code.

Reply:
TBH I didn't comment because I don't have more thoughts about it, and am a bit wishy-washy on it and didn't want to continue to contradict myself.

Pretty sure we can still just `echo`; if something is installed on the server, the only thing that needs to be in there is `da_install_type`. `da_project_name` is entirely unused, and `artifacts_path` will be added when ALKiln runs the tests, it doesn't have to be there before.

`dainstall` is fewer lines, but it's working differently (by zipping up an on-machine directory instead of from a repository), and does add a new python dependency to things. It's practical, but not clean IMO. `dainstall` is the thing that makes it "not portable", [like we've talked about in other places](https://github.com/SuffolkLITLab/ALKiln/pull/753#discussion_r1300557592). Idk, not arguing for anything in particular here.

I'm starting to think that passing the base URL (or what I'd call the URL path, from the end of the domain to the beginning of the interview/) directly isn't a good idea; it's much more complicated and doesn't add any new possibilities for people, and that sticking with `install_method` has benefits here; it only gets rid of like, 5 lines of JS?

Reply:
You have some good points. If we're going to `echo` in the action then we still need to have the js handle a situation that isn't a GitHub action. Do you have preferences for how to handle that? We can just use session vars defaults with a non-env var. That's part of why we called it `session_vars` as it does more than just env vars.

Just a note for Future People - `da_project_name` is used locally, but that's not relevant in the GitHub action.

Reply:
Possibly a conclusion: I think now that we're going to start using `dainstall` for GitHub server installation and _also_ use it for remote server Playground installation (to be able to install PRs and such), either we need to keep the old setup scripts for local developers or we need to do a good job documenting how local devs should use the command line to install stuff on remote servers.